src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/CastHelpers.cs

```diff
@@ -24,6 +25,9 @@ internal static void ThrowInvalidCastException(object fromType, void* toTypeHnd)
             throw null!; // Provide hint to the inliner that this method does not return
         }
 
+        [DoesNotReturn]
+        internal static void ThrowInvalidCastException() => throw new InvalidCastException();
```

Comment:
Yes, it can be private. The method above this is `internal static void ThrowInvalidCastException(object fromType, void* toTypeHnd)`. Just added an overload. The other overload makes a QCall and ends up using `COMPlusThrowInvalidCastException`.

```suggestion
        [DoesNotReturn]
        private static void ThrowInvalidCastException() => throw new InvalidCastException();
```